File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    result = set()
    result.add(words[0])
    
    for i in range(1,len(words)):
        if words[i-1][-1] != words[i][0] or words[i] in result:
            return [i%n+1,i//n+1] 
        
        result.add(words[i])
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] result = {0, 0};
        Set<String> set = new HashSet<>();
        set.add(words[0]);

        for (int i = 1; i < words.length; i++) {
            String prev = words[i - 1];
            String curr = words[i];

            // 끝말잇기 실패
            if (prev.charAt(prev.length() - 1) != curr.charAt(0) || set.contains(curr)) {
                result[0] = (i % n) + 1;  // 사람 번호
                result[1] = (i / n) + 1;  // 차례
                return result;
            }

            set.add(curr);
        }

        return result;  // 규칙 위반 없을 경우 [0, 0]
    }
}